cmd/blockchaincmd/deploy.go

```diff
@@ -1145,58 +1154,78 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
-				Amount:             relayerAmount,
-				AllowPrivateIPs:    relayerAllowPrivateIPs,
-			}
-			if network.Kind == models.Local || useLocalMachine {
-				deployRelayerFlags.Key = constants.ICMRelayerKeyName
-				deployRelayerFlags.Amount = constants.DefaultRelayerAmount
-				deployRelayerFlags.BlockchainFundingKey = constants.ICMKeyName
-			}
-			if network.Kind == models.Local {
-				deployRelayerFlags.CChainFundingKey = "ewoq"
 			}
-			if err := relayercmd.CallDeploy(nil, deployRelayerFlags, network); err != nil {
-				ux.Logger.PrintToUser("Relayer is not deployed due to %v", err)
-				ux.Logger.PrintToUser("To deploy relayer, call `avalanche interchain relayer deploy`")
-			} else {
-				ux.Logger.GreenCheckmarkToUser("Relayer is successfully deployed")
+			if !icmSpec.SkipRelayerDeploy && network.Kind != models.Mainnet {
+				deployRelayerFlags := relayercmd.DeployFlags{
+					Version:            icmSpec.RelayerVersion,
+					BinPath:            icmSpec.RelayerBinPath,
+					LogLevel:           icmSpec.RelayerLogLevel,
+					RelayCChain:        relayCChain,
+					CChainFundingKey:   cChainFundingKey,
+					BlockchainsToRelay: []string{blockchainName},
+					Key:                relayerKeyName,
+					Amount:             relayerAmount,
+					AllowPrivateIPs:    relayerAllowPrivateIPs,
+				}
+				if network.Kind == models.Local || useLocalMachine {
+					deployRelayerFlags.Key = constants.ICMRelayerKeyName
+					deployRelayerFlags.Amount = constants.DefaultRelayerAmount
+					deployRelayerFlags.BlockchainFundingKey = constants.ICMKeyName
+				}
+				if network.Kind == models.Local {
+					deployRelayerFlags.CChainFundingKey = "ewoq"
+					deployRelayerFlags.CChainAmount = constants.DefaultRelayerAmount
+				}
+				if err := relayercmd.CallDeploy(nil, deployRelayerFlags, network); err != nil {
+					relayerErr = err
+					ux.Logger.RedXToUser("Relayer is not deployed due to: %v", relayerErr)
+				} else {
+					ux.Logger.GreenCheckmarkToUser("Relayer is successfully deployed")
+				}
 			}
 		}
 	}
-	if sidecar.Sovereign {
-		ux.Logger.GreenCheckmarkToUser("L1 is successfully deployed on %s", network.Name())
-	} else {
-		ux.Logger.GreenCheckmarkToUser("Subnet is successfully deployed %s", network.Name())
-	}
+
 	flags := make(map[string]string)
 	flags[constants.MetricsNetwork] = network.Name()
 	metrics.HandleTracking(cmd, constants.MetricsSubnetDeployCommand, app, flags)
 
 	if network.Kind == models.Local && !simulatedPublicNetwork() {
 		ux.Logger.PrintToUser("")
-		return PrintSubnetInfo(blockchainName, true)
+		_ = PrintSubnetInfo(blockchainName, true)
+	}
+
+	if icmErr != nil {
+		ux.Logger.PrintToUser("")
+		ux.Logger.PrintToUser("Interchain Messaging is not deployed due to: %v", icmErr)
+		ux.Logger.PrintToUser("")
+		ux.Logger.PrintToUser("To deploy ICM later on, call `avalanche icm deploy`")
+		ux.Logger.PrintToUser("To deploy a local relayer later on, call `avalanche interchain relayer deploy`")
```

Comment:
are you sure you want to display `avalanche interchain relayer deploy` hint for icm case ?